### DISCUSSION TITLE
the instance list is gone when i delete yesterday data

### DISCUSSION THREAD

**Discussion by taotoxht** [apache/skywalking#9786] [Q&A] [answered]
Dear Team,

I am use skywalking on production, Because the elasticsearch disk is full, so i write a linux crontab job delete yesterday data every day at 6：00 pm。then the instance list is gone from the Instance tab of General-Service dashboard。 like below screenshot

![image](https://user-images.githubusercontent.com/16130118/195737250-c5d42cd0-ad17-4841-82ba-57864050f2dd.png)

in the overview tab the data is normal：

![image](https://user-images.githubusercontent.com/16130118/195737414-80073ebe-7678-4b46-87e8-b23f0083d4f4.png)

After waiting all day, it didn't recover ， must i restart the client agent, it could be ok? but i  am on production env, Is there any other way to solve it?

Thanks and Regards,

**Comment by wu-sheng** [comment, marked as answer]
Are you making the crontab keep running? Ideally, the instance would be regenerated if it has traffic/metrics.

More importantly, removing yesterday's data is a very bad practice. Don't do that. 

You could reboot the whole OAP server cluster, and I believe the data would be bad. Of course, make sure you would not create a crontab like this.

**Comment by taotoxht** [comment]
since I reboot the oap , i recoverd.

**Comment by taotoxht** [comment]
I'm curious why the heart beat exists,why it can't regenerate instance_traffic. I found from this page below：
https://github.com/apache/skywalking/issues/3453
![image](https://user-images.githubusercontent.com/16130118/195754543-13bc6474-9fe5-445a-b242-eac3773797ce.png)
I stop the crontab and  waiting all day，I thought it would recover automatically，but it didn't 。 thank you~

**Reply by wu-sheng** [reply]
I don't know either. You may remove the index template, rather than the indices. But I am not guessing.

**Comment by taotoxht** [comment]
My crontab job execute below shell：
curl --location --request DELETE 'http://10.153.99.7:9200/*-20221011*' --header 'Accept: application/json'  --header 'Authorization: Basic ZWxhc3RpYzpnaHdjdUVzVj13U3lrLV9WWF9GNw==' --header 'Cache-Control: no-cache' --header 'Content-Type: application/json' 
does this can delete  oap es temlapte？
but i can't found  oap elastic temlpate by es api：http://10.153.99.7:19200/_template， only this
![image](https://user-images.githubusercontent.com/16130118/195759574-120c6c79-f40d-4557-aaf6-219c0a34368c.png)
 thank you~

**Reply by wu-sheng** [reply]
I can't tell what happens on your site, but SkyWalking definitely has index templates.

**Reply by taotoxht** [reply]
i kwnow why i can't find the index template.  I change 'http://10.153.99.7:19200/_template' to 'http://10.153.99.7:9200/_index_template/xht01*' ,then get the result as below:
![image](https://user-images.githubusercontent.com/16130118/195784112-4f6f6cbb-9553-4dc5-93bb-092df9195f6e.png)

but i still don't know why the instance not regenerated ，i am sure there is a lot of traffic  as shown in the screenshot above

**Comment by sruby** [comment]
I had the same problem because our operator cleared ES data through crontab jobs.
Fix options:
Stop crontab jobs, stop SkyWalking, clear ES data and start SkyWalking.
